Microservices/user-service/src/main/java/app/sportahub/userservice/service/auth/AuthServiceImpl.java

```diff
@@ -88,11 +89,17 @@ public LoginResponse loginUser(LoginRequest loginRequest) {
 
     @Override
     public TokenResponse refreshToken(RefreshTokenRequest tokenRequest) {
-
         return Mono.from(keycloakApiClient.refreshToken(tokenRequest.refreshToken())).map(jsonNode -> {
             String refreshToken = jsonNode.get("refresh_token").asText();
             String accessToken = jsonNode.get("access_token").asText();
             return new TokenResponse(accessToken, refreshToken);
         }).block();
     }
+
+    @Override
+    public void sendVerificationEmail(String email) {
+        User user = userRepository.findUserByEmail(email).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found"));
```

Comment:
In here, use the custom exception we have created for that purpose. Refer to this [link](https://github.com/SOEN490-SportsApp/SportsApp/blob/50d5dd3eb37053f5930247c58b1a57751bc5a4e5/Microservices/user-service/src/main/java/app/sportahub/userservice/service/user/UserServiceImpl.java#L66-L67) for an example use of the exception 👍